src/utils/daos_hdlr.c

```diff
@@ -3020,50 +3071,10 @@ fs_copy_hdlr(struct cmd_args_s *ap)
 		D_GOTO(out, rc = daos_errno2der(rc));
 	}
 
-	parse_filename_dfs(src_str, &name, &dname);
-
-	/* construct destination directory in DAOS, this needs
-	 * to strip the dirname and only use the basename that is
-	 * specified in the dst argument
-	 */
-	src_str_len = strlen(dname);
-	D_ASPRINTF(dst_dir, "%s/%s", dst_str, src_str + src_str_len);
-	if (dst_dir == NULL) {
-		fprintf(stderr,
-			"Unable to allocate memory for destination path.\n");
-		D_GOTO(out_disconnect, rc = -DER_NOMEM);
-	}
-	/* set paths based on file type for source and destination */
-	if (src_file_dfs.type == POSIX && dst_file_dfs.type == DAOS) {
```

Comment:
@daltonbohning ok, so making sure my understanding of this is correct... fs_copy is not a switch statement to decide which type of copy to be performed instead of doing it here in fs_copy_hdlr? And the functionality in fs_copy was moved/split into fs_copy_dir and fs_copy_file?

Reply:
@dsikich
> fs_copy is now a switch statement to decide which type of copy to be performed

Not exactly. The previous logic checked explicitly for DAOS->DAOS, DAOS->POSIX, or POSIX->DAOS. I removed those explicit checks so now we just generally copy src_file_dfs to dst_file_dfs (either of which can be POSIX or DAOS). If you look at the logic closely for each of the previous if statements, it was mostly the same for each. The mkdir call was moved to fs_copy because we only need to call mkdir if we are copying to a directory, but now a destination file is supported as well.
The switch in `fs_copy` is not for the _**type**_ of copy in terms of DAOS or POSIX, but rather for whether the src and dst are directories or files.

> the functionality in fs_copy was moved/split into fs_copy_dir and fs_copy_file

Right, for a few reasons:
1. There was a need to call fs_copy_dir in multiple places.
2. If/when we add support for other types (symlinks) we can just add another function for that case.
3. I was emulating DFS somewhat in that it also has switch statements for types.

Reply:
@daltonbohning ok, I suppose I wasn't very clear I was thinking of "type" as copying a directory vs. a file. I see that you removed the checks for DAOS->DAOS, DAOS->POSIX, etc. which makes sense. As I was writing the tool and figuring it out I used them to break up the implementation, and it changed enough over time that they were mostly the same by the end as you pointed out.

Reply:
@dsikich Right. Most of the support for the generics was already there. I just bumped the 99% to 100% :)